### DISCUSSION TITLE
Unmount if hidden

### DISCUSSION THREAD

**Discussion by lanfr0** [lodev09/react-native-true-sheet#74] [Ideas]
Hi, I'm questioning myself whether unmounting the component when it is hidden could be useful. I mean, in some cases I would like to achieve a stateless behaviour or simply use a single Portal to render the current active bottom sheet.
I tested that If I use two or more bottom sheet with the same Portal, only the last one is displayed in the view.

What do you think about?

**Comment by lodev09** [comment]
It's up to you. You can do that if you want. Just treat the sheet as a standard view container. It will still be up to your content/views that may cause performance issue -- not the sheet itself.

On my case, I have a wrapper component for TrueSheet and I'm only rendering it when the screen is shown (react-navigation). See [this guide](https://sheet.lodev09.com/troubleshooting#present-during-mount) for reference.

**Reply by wcastand** [reply]
is there a way to get a listener or something to only render when the sheet is presented? or i have to do it manually every time?

**Reply by lodev09** [reply]
It will look weird if you render it after the presentation animation. You want it to render in the "background" first for smooth transition. I think the only concern on your case is to render it when "needed" to present, hence my comment above

**Comment by lodev09** [comment]
v3 now lazy loads content by default! ❤️
